api/src/codec-to-json.ts

```diff
@@ -0,0 +1,47 @@
+import { bytes, codec, context } from "@typeberry/block";
+import type { FastifyReply, FastifyRequest } from "fastify";
+import { createErrorResponse } from "./error";
+import { kinds } from "./kinds";
+
+type Req = { Params: { id: string }; Body: { codec: string } };
+type SuccessRes = { data: { json: string } };
+type Res = { Body: SuccessRes };
+
+export function codecToJson(req: FastifyRequest<Req>, reply: FastifyReply<Res>) {
+  const descriptorId = req.params.id;
+  const descriptor = kinds.find((x) => x.name === descriptorId)?.clazz;
+  if (!descriptor) {
+    reply.status(400).send(createErrorResponse("Wrong descriptor id", `Descriptor ${descriptorId} does not exist.`));
+    return;
+  }
+
+  const decoded = codec.Decoder.decodeObject(
+    descriptor.Codec,
+    bytes.BytesBlob.parseBlob(req.body.codec),
```

Comment:
Same question here about validation. Are we guaranteed to have `body.codec` here?

Reply:
ditto